src/main/java/com/cloudcoreo/plugins/jenkins/CloudCoreoPublisher.java

```diff
@@ -0,0 +1,224 @@
+    private CloudCoreoTeam team;
+
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    public boolean getBlockOnHigh() {
+        return blockOnHigh;
+    }
+
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    public boolean getBlockOnMedium() {
+        return blockOnMedium;
+    }
+
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    public boolean getBlockOnLow() {
+        return blockOnLow;
+    }
+
+    CloudCoreoTeam getTeam() {
+        return team;
+    }
+
+    @SuppressWarnings("WeakerAccess")
+    @DataBoundConstructor
+    public CloudCoreoPublisher(boolean blockOnHigh, boolean blockOnMedium, boolean blockOnLow) {
+        this.blockOnHigh = blockOnHigh;
+        this.blockOnMedium = blockOnMedium;
+        this.blockOnLow = blockOnLow;
+        workspacePath = null;
+    }
+
+    @Override
+    public Action getProjectAction(AbstractProject<?, ?> project) {
+        return new CloudCoreoProjectAction(project);
+    }
+
+    @Extension
+    public static class DescriptorImpl extends BuildStepDescriptor<Publisher> {
+        @Nonnull
+        @Override
+        public String getDisplayName() {
+            return "Analyze CloudCoreo Results";
+        }
+
+        public boolean isApplicable(Class jobType) {
+            return true;
+        }
+
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) super.getDescriptor();
+    }
+
+
+    private Map<String, String> readSerializedDataFromTempFile(String buildId)
+            throws URISyntaxException, IOException, ClassNotFoundException {
+        String fp = "file:///" + workspacePath + "/" + buildId + ".ser";
+        File file = new File(new URI(fp.replaceAll(" ", "%20")).getPath());
+        FileInputStream f = new FileInputStream(file);
+        ObjectInputStream objectStream = new ObjectInputStream(f);
+
+        @SuppressWarnings("unchecked")
+        Map<String, String> vars = (Map<String, String>) objectStream.readObject();
+
+        objectStream.close();
+        return vars;
+    }
+
+    @SuppressWarnings("NullableProblems")
+    @Override
+    public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener) {
+
+        logger = listener.getLogger();
+        workspacePath = workspace;
+        ResultManager resultManager = new ResultManager(blockOnLow, blockOnMedium, blockOnHigh, logger);
+        if (!resultManager.shouldBlockBuild()) {
+            // no blocking for failures requested
+            return;
+        }
+
+        try {
+            initializeTeam(build);
+            waitForContextRun(build);
+            retrieveAndSetResults(resultManager, build.getId());
+
+            writeResults(build, resultManager);
+            if (resultManager.hasBlockingFailures()) {
+                resultManager.reportResultsToConsole();
+                build.setResult(Result.FAILURE);
+            }
+        } catch (ExecutionFailedException ignore) {}
```

Comment:
All of the above throw `ExecutionFailedException` when we error out on something and we don't want to block the build.